### PR TITLE
Issue 10 expose list of artificats

### LINKED ISSUE #10: Expose list of Artifact versions via a paginated collection
I should be able to request versions for a (`groupName`, `artifactName`) tuple in the REST API and then paginate through them similar to [GitHub's API](https://developer.github.com/v3/#pagination).

Artifact version  get request will be modified to specify the page number and per page size. These parameters are optional, if these parameters are missing, the get request will default to page size 0 and per page size to 5.

### PR DESCRIPTION
**Pull request by GiriDandu**
Fix #10

### CODE REVIEW

File: features/api/artifact/reading.feature [hunk 1]
```diff
@@ -42,27 +42,48 @@ Feature: Artifact READ APIs
...
+    When I GET "/api/v1/artifacts/by-module/com.example.cucumber:cucumber-artifact?pageNumber=1&perPageSize=5"
+    Then the response should be 404
+
 
   @wip
   Scenario: Fetching an artifact's versions
 
     Given there is an artifact
-    When I GET "/api/v1/artifacts/1/versions"
+    When I GET "/api/v1/artifacts/1/versions
```
Comment (rtyler) [inline features/api/artifact/reading.feature]: There's a quote missing here which is incorrect

Comment (GiriDandu) [inline features/api/artifact/reading.feature]: Will fix that

File: features/api/artifact/reading.feature [hunk 2]
```diff
@@ -42,21 +42,42 @@ Feature: Artifact READ APIs
...
   Scenario: Fetching an artifact by the (group, name) tuple that doesn't exist
 
     When I GET "/api/v1/artifacts/by-module/com.example.cucumber:cucumber-artifact"
     Then the response should be 404
 
+Scenario: Fetching an artifact by the (group, name) tuple with invalid page number
+
+    When I GET "/api/v1/artifacts/by-module/com.example.cucumber:cucumber-artifact?pageNumber=1&perPageSize=5"
+    Then the response should be 404
+
```
Comment (maheshkelkar) [inline features/api/artifact/reading.feature]: could you please add a testcases for invalid page number (negative number or string), invalid page size?

Comment (GiriDandu) [inline features/api/artifact/reading.feature]: Added one more test case for invalid data type

File: features/api/artifact/reading.feature [hunk 3]
```diff
@@ -42,21 +42,46 @@ Feature: Artifact READ APIs
...
+        "version" : "1.0.1",
+        "sourceUrl" : "http://example.com/maven/com.example.cucumber/cucumber-artifact/1.0.1/cucumber-artifact-1.0.1.jar"
+      }]
+    """
   Scenario: Fetching an artifact by the (group, name) tuple that doesn't exist
 
     When I GET "/api/v1/artifacts/by-module/com.example.cucumber:cucumber-artifact"
     Then the response should be 404
 
+  Scenario: Fetching an artifact by the (group, name) tuple with invalid page number
```
Comment (rtyler) [inline features/api/artifact/reading.feature]: Will you annotate this with the `@error` tag

File: features/api/artifact/reading.feature [hunk 4]
```diff
@@ -42,21 +42,46 @@ Feature: Artifact READ APIs
...
+
+    When I GET "/api/v1/artifacts/by-module/com.example.cucumber:cucumber-artifact?pageNumber=1&perPageSize=5"
+    Then the response should be 404
+
+  Scenario: Fetching an artifact by the (group, name) tuple with invalid data type for pageSize
```
Comment (rtyler) [inline features/api/artifact/reading.feature]: Will you annotate this with the `@error` tag

**Comment by rtyler** [comment]
Overall this looks really good, I'm going to go ahead and merge it and perhaps a follow up commit could include the tags requested in the commit comments

**Pull request merged**